lib/diaspora/federation/dispatcher.rb

```diff
@@ -0,0 +1,84 @@
+module Diaspora
+  module Federation
+    class Dispatcher
+      include Diaspora::Logging
+
+      def initialize(sender, object, opts={})
+        @sender = sender
+        @object = object
+        @opts = opts
+      end
+
+      def self.build(sender, object, opts={})
+        if object_should_be_processed_as_public?(object)
+          Public.new(sender, object, opts)
+        else
+          Private.new(sender, object, opts)
+        end
+      end
+
+      def self.defer_dispatch(sender, object, opts={})
+        object.save if object.new_record?
```

Comment:
Why the explicit check, is there a reason why it shouldn't be saved if there are just changed attributes? Otherwise `save` is a noop if the object is `persisted? && !changed?`